zeq/http/server.h

```diff
@@ -0,0 +1,120 @@
+
+/* Copyright (c) 2016, Human Brain Project
+ *                     Stefan.Eilemann@epfl.ch
+ */
+
+#ifndef ZEQ_HTTP_SERVER_H
+#define ZEQ_HTTP_SERVER_H
+
+#include <zeq/receiver.h> // base class
+
+namespace zeq
+{
+/** HTTP protocol support. */
+namespace http
+{
+/**
+ * Serves HTTP GET and POST requests for servus::Serializable objects.
+ *
+ * Behaves semantically like a Publisher (for GET) and Subscriber (for POST),
+ * except uses HTTP with JSON payload as the protocol. Requests are served
+ * synchronously (as per HTTP spec).
+ *
+ * Not thread safe.
+ *
+ * Example: @include tests/http/server.cpp
+ */
+class Server : public zeq::Receiver
+{
+public:
+    /**
+     * Construct a new HTTP server.
+     *
+     * To process requests on the incoming port, call receive().
+     *
+     * @param uri The server address in the form "[tcp://][hostname][:port]"
+     * @param shared a shared receiver, see Receiver constructor.
+     * @throw std::runtime_error on malformed URI or connection issues.
+     */
+    ZEQ_API Server( const URI& uri, Receiver& shared );
+    ZEQ_API explicit Server( const URI& uri );
+    ZEQ_API explicit Server( Receiver& shared );
+    ZEQ_API explicit Server();
```

Comment:
all ctors should have their own documentation, at least the default ctor that it runs on localhost:4020 for instance.